include/internal/sockets.h

```diff
@@ -132,39 +132,39 @@ struct servent *PASCAL getservbyname(const char *, const char *);
 #  define get_last_socket_error() WSAGetLastError()
 #  define clear_socket_error()    WSASetLastError(0)
 #  define readsocket(s,b,n)       recv((s),(b),(n),0)
-#  define writesocket(s,b,n)      send((s),(b),(n),0)
+#  define writesocket(s,b,n,f)    send((s),(b),(n),(f))
 # elif defined(__DJGPP__)
 #  define WATT32
 #  define WATT32_NO_OLDIES
 #  define closesocket(s)          close_s(s)
 #  define readsocket(s,b,n)       read_s(s,b,n)
-#  define writesocket(s,b,n)      send(s,b,n,0)
+#  define writesocket(s,b,n,f)    send(s,b,n,f)
 # elif defined(OPENSSL_SYS_VMS)
 #  define ioctlsocket(a,b,c)      ioctl(a,b,c)
 #  define closesocket(s)          close(s)
 #  define readsocket(s,b,n)       recv((s),(b),(n),0)
-#  define writesocket(s,b,n)      send((s),(b),(n),0)
+#  define writesocket(s,b,n,f)    send((s),(b),(n),(f))
 # elif defined(OPENSSL_SYS_VXWORKS)
 #  define ioctlsocket(a,b,c)          ioctl((a),(b),(int)(c))
 #  define closesocket(s)              close(s)
 #  define readsocket(s,b,n)           read((s),(b),(n))
-#  define writesocket(s,b,n)          write((s),(char *)(b),(n))
+#  define writesocket(s,b,n,f)        write((s),(char *)(b),(n))
```

Comment:
> Should it be mentioned that for some of these supported platforms, that the flags aren't available? Should this be changed from a `write` to a `send`? (I haven't had to deal with VXWORKS in a while, but I would think that they might have a `send` socket API.) Same goes for a few other OS's here.

I think the better approach would be for any one familiar with VxWORKS or that have access to those platforms that can test it out to comment back if they have a send() call that takes socket flags. I do think that every BSD sockets implementation should have it but since the old code used write() specifically for these platforms I got cold feet and left them as is.